cluster/gce/trusty/node.yaml

```diff
@@ -372,5 +372,45 @@ script
 	fi
 end script
 
+--===============6024533374511606659==
+MIME-Version: 1.0
+Content-Type: text/upstart-job; charset="us-ascii"
+Content-Transfer-Encoding: 7bit
+Content-Disposition: attachment; filename="kube-node-health-monitoring.conf"
+
+description "Kubenetes node health monitoring"
+
+start on stopped kube-docker and started kube-proxy
+
+respawn
+
+script
+	set -o nounset
+
+	# Wait for a minute to let docker, kubelet, and kube-proxy processes finish initialization.
+	# TODO(andyzheng0831): replace it with a more reliable method if possible.
+	sleep 60
+
+	sleep_seconds=10
+	max_seconds=10
+	# We simply kill the process when there is a failure. Another upstart job will automatically
+	# restart the process.
+	while [ 1 ]; do
+		if ! timeout 10 docker version > /dev/null; then
+			echo "Docker daemon failed!"
+			pkill docker
+		fi
+		if ! curl -m ${max_seconds} -f -s http://127.0.0.1:10255/healthz > /dev/null; then
```

Comment:
I'm planning to switch the kubelet healthchecking to use the secure port (see https://github.com/kubernetes/kubernetes/pull/13044) rather than 10255. Remind me to change this file as well once that PR is merged.